### DISCUSSION TITLE
how to show tool tip for disabled tool

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2150] [Q&A]
### Introduction

![clear paper](https://user-images.githubusercontent.com/117301232/234487384-1c8e0109-ec8c-4450-b76a-8d6bd0100c71.png)

when I enter the canvas page the undo redo buttons are disabled. (it will enable when we make any changes in the paper)  when I hover the disabled button I need to show the tooltips, is there any way to show tooltips for the disabled button

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Disabled elements do not trigger events apparently ([StackOverflow](https://stackoverflow.com/questions/18113937/fire-onmouseover-event-when-element-is-disabled)).

1. You can set `autoToggle: false` to both buttons, so they will never get disabled
```js
{ 
  type: 'redo',
  autoToggle: false,
  attrs: {
      button: {
          'data-tooltip': 'JJ+ Tooltip',
      }
   }
}
```
2. You can add `title` attribute 
```js
{ 
  type: 'redo',
  attrs: {
      button: {
          'title': 'Native Tooltip',
      }
   }
}
```
